pkg/apis/jvmbuildservice/v1alpha1/artifactbuildrequest_types.go

```diff
@@ -7,17 +7,20 @@ import (
 type ArtifactBuildRequestSpec struct {
 	// GAV is the groupID:artifacdtID:version tuple seen in maven pom.xml files
 	GAV string `json:"gav,omitempty"`
+	Message           string `json:"message,omitempty"`
+	SCMURL            string `json:"scmURL,omitempty"`
```

Comment:
It does feel like this change means our implementation is now leaking out into the CRD, as conceptually these really are part of the status, they are not things that a user would update.

Reply:
yeah admittedly I'm back and forth a bit on these vectors of consideration:
- which reconcilers maintain which fields, and do we need to adhere to the CRD specific reconciler conventions. 
- do these CRDs, this data model, truly fit the "spec/status" paradigm ... I can point you to serveral "request" types in the openshift ecosystem that do *NOT* have spec/status.  For example:  the [openshift build request](https://github.com/openshift/api/blob/master/build/v1/types.go#L1247-L1280) ... I wonder if the ABR for example is conceptually more akin to that.  If so, we could split things out for example, and have an `ArtifactBuildRequest` that results in the creation of an `ArtifactBuild`
- who exactly are the "users" of these CRD specifically (vs. the overall system in general) and how do they interact with the system.  Even after watching your demo videos, I'm not 100% sure on the answers to that yet.

That said, I think all that ^^ dovetails precisely into the theme of tomorrow's next workshop that @sbose78 has set up.  If anything else, I think this PR could kick start discussion on the CRD API, where we will get to consensus one way or the other, and even if we throw this PR away, as long as we get clarity and consensus, the greater purpose has been served.